$OpenBSD: patch-libarchive_archive_read_c,v 1.1 2016/01/30 23:34:44 naddy Exp $

Commit e6c9668
    Add a check to archive_read_filter_consume to reject any
    attempts to move the file pointer by a negative amount.

--- libarchive/archive_read.c.orig	Fri Feb  8 04:24:42 2013
+++ libarchive/archive_read.c	Sat Jan 30 21:40:21 2016
@@ -1394,6 +1394,8 @@ __archive_read_filter_consume(struct archive_read_filt
 {
 	int64_t skipped;
 
+	if (request < 0)
+		return ARCHIVE_FATAL;
 	if (request == 0)
 		return 0;
 
